### ISSUE TITLE
Resolver plugins

### ISSUE THREAD

**Issue by shermendev** [benmosher/eslint-plugin-import#1628]
Is there a way to add a resolver plugin to eslint or I will have to disable the `import/no-unresolved` rule?
Here's a plugin I use:
```jsx
const DirectoryNamedWebpackPlugin = require('directory-named-webpack-plugin')

module.exports = {
  resolve: {
    plugins: [
      new DirectoryNamedWebpackPlugin(true)
    ]
  }
}
```
Tried this but it didn't work
```jsx
//.eslintrc.js
module.exports = {
  settings: {
    'import/resolver': {
      plugins: [new DirectoryNamedWebpackPlugin(true)]
    }
  }
}
```

**Comment by ljharb** [comment]
There is - https://github.com/benmosher/eslint-plugin-import#resolvers - but there’s also a webpack resolver specifically: https://npmjs.com/eslint-import-resolver-webpack

**Comment by shermendev** [comment]
In https://github.com/benmosher/eslint-plugin-import#resolvers I can't seem to find a way to use a plugin(perhaps I'm missing something)? 
I also can't use `eslint-import-resolver-webpack` because of [craco](https://github.com/gsoft-inc/craco) (I tried it without craco but it didn't catch the plugin as well)

**Comment by ljharb** [comment]
I’m not sure why CRA would affect anything; the webpack resolver just reads your local webpack config.

A webpack plugin certainly can’t be dropped in to resolver settings; its an entirely different thing. You could build your own resolver, however.

**Comment by shermendev** [comment]
> 
> A webpack plugin certainly can’t be dropped in to resolver settings;

I see, thank you for clarifying that out.

**Comment by egehandulger** [comment]
> In https://github.com/benmosher/eslint-plugin-import#resolvers I can't seem to find a way to use a plugin(perhaps I'm missing something)?
> I also can't use `eslint-import-resolver-webpack` because of [craco](https://github.com/gsoft-inc/craco) (I tried it without craco but it didn't catch the plugin as well)

Anyone who comes here in the future. You certainly can use `eslint-import-resolver-webpack`. In the readme it mentions about this:

```
Will look for webpack.config.js as a sibling of the first ancestral package.json, or a config parameter may be provided with another filename/path either relative to the package.json, or a complete, absolute path.
```

Clearly, `craco.config.js` is not the same as `webpack.config.js`. To make both modules happy, create a `webpack.config.js` file as if there is no `craco`.

```
const DirectoryNamedWebpackPlugin = require('directory-named-webpack-plugin')

module.exports = {
  resolve: {
    plugins: [new DirectoryNamedWebpackPlugin()],
  },
}
```

Then, require this file inside `craco.config.js` and use it where applicable/desired in accordance to `craco` format.

```
const webpackCustomConfig = require('./webpack.config')

module.exports = {
  webpack: {
    configure: (webpackConfig) => ({
      ...webpackConfig,
      resolve: {
        ...webpackConfig.resolve,
        plugins: [
          ...webpackCustomConfig.resolve.plugins,
          ...webpackConfig.resolve.plugins,
        ],
      },
    }),
  },
}
```

For `eslint`, you can configure `webpack` as documented in the readme:
```
settings: {
    'import/resolver': 'webpack',
  },
```
